packages/wrapper-react/src/index.tsx

```diff
@@ -30,72 +29,49 @@ export const MonacoEditorReactComp: React.FC<MonacoEditorProps> = (props) => {
+        const destroyMonaco = async () => {
+            try {
+                await wrapperRef.current.dispose();
+            } catch {
+                // The language client may throw an error during disposal.
+                // This should not prevent us from continue working.
+            }
         };
-    }, []);
-
-    useEffect(() => {
-        handleReInit();
-    }, [wrapperConfig]);
-
-    useEffect(() => {
-        handleOnTextChanged();
-    }, [onTextChanged]);
-
-    useEffect(() => {
-        if (containerRef.current) {
-            containerRef.current.className = className ?? '';
-            wrapperConfig.htmlContainer = containerRef.current;
-        }
-    }, [className]);
 
-    const handleReInit = useCallback(async () => {
-        await destroyMonaco();
-        await initMonaco();
-        await startMonaco();
-    }, [wrapperConfig]);
-
-    const initMonaco = useCallback(async () => {
-        if (containerRef.current) {
-            wrapperConfig.htmlContainer = containerRef.current;
-            await wrapperRef.current.init(wrapperConfig);
-        } else {
-            throw new Error('No htmlContainer found! Aborting...');
-        }
-    }, [wrapperConfig]);
+        const initMonaco = async () => {
+            if (containerRef.current) {
+                wrapperConfig.htmlContainer = containerRef.current;
+                await wrapperRef.current.init(wrapperConfig);
+            } else {
+                throw new Error('No htmlContainer found! Aborting...');
+            }
+        };
 
-    const startMonaco = useCallback(async () => {
-        if (containerRef.current) {
-            try {
-                wrapperRef.current.registerTextChangeCallback(onTextChanged);
-                await wrapperRef.current.start();
-                onLoad?.(wrapperRef.current);
-                handleOnTextChanged();
-            } catch (e) {
-                if (onError) {
-                    onError(e);
-                } else {
-                    throw e;
+        const startMonaco = async () => {
+            if (containerRef.current) {
+                try {
+                    wrapperRef.current.registerTextChangeCallback(onTextChanged);
+                    await wrapperRef.current.start();
+                    onLoad?.(wrapperRef.current);
+                } catch (e) {
+                    if (onError) {
+                        onError(e);
+                    } else {
+                        throw e;
+                    }
                 }
+            } else {
+                throw new Error('No htmlContainer found! Aborting...');
             }
-        } else {
-            throw new Error('No htmlContainer found! Aborting...');
-        }
-    }, [onError, onLoad, onTextChanged]);
+        };
 
-    const handleOnTextChanged = useCallback(() => {
-        if (!onTextChanged) return;
-    }, [onTextChanged, wrapperConfig]);
+        (async () => {
+            await destroyMonaco();
```

Comment:
I know this was like this before, but it makes no sense, because if you have the second use effect for destruction, the init of the wrapper ensures everything was disposed.

Reply:
The above code snippet (second reply to first comment) takes this into account already.